jgiven-core/src/main/java/com/tngtech/jgiven/impl/ScenarioModelBuilder.java

```diff
@@ -141,6 +141,24 @@ public void introWordAdded( String value ) {
         introWord.setValue( value );
     }
 
+    @Override
+    public void stepCommentAdded( List<NamedArgument> arguments ) {
+        if( arguments == null || arguments.size() != 1 ) {
+            throw new JGivenWrongUsageException( "A step comment method must have exactly one parameter." );
+        }
+
+        if( !( arguments.get( 0 ).getValue() instanceof String ) ) {
+            throw new JGivenWrongUsageException( "The step comment method parameter must be a string." );
```

Comment:
One could indeed. But is there a usecase for it? :-) Seems like a non-string comment is actually more of an attachment.